core/tx_list.go

```diff
@@ -123,6 +125,36 @@ func (m *txSortedMap) Filter(filter func(types.PoolTransaction) bool) types.Pool
 	return removed
 }
 
+// Filter iterates over the list of transactions and removes all of them for which
+// the specified function evaluates to true. Return the filtered transactions and
+// corresponding errors.
+func (m *txSortedMap) FilterWithError(filter func(types.PoolTransaction) error) (types.PoolTransactions, []error) {
+	var (
+		removed types.PoolTransactions
+		errs    []error
+	)
+
+	// Collect all the transactions to filter out
+	for nonce, tx := range m.items {
+		if err := filter(tx); err != nil {
+			removed = append(removed, tx)
+			errs = append(errs, err)
+			delete(m.items, nonce)
+		}
+	}
+	// If transactions were removed, the heap and cache are ruined
```

Comment:
what does this mean？

Reply:
This is the comment from copying `txSortedMap.Filter`. It means when transactions are moved, the order of heap is destroyed. Thus, will need to re-build `m.index` and `m.items` in the following function